### PR TITLE
feat: TypeScript 5.6 and `npm:@types/node@22`

### LINKED ISSUE #24238: "".isWellFormed() doesn't type-check
Version: Deno 1.44.2

The isWellFormed() method on strings works, but doesn't type-check.

```
% cat >> bug.ts
"".isWellFormed()

% deno check bug.ts 
Check [redacted]/bug.ts
error: TS2339 [ERROR]: Property 'isWellFormed' does not exist on type '""'.
"".isWellFormed()
   ~~~~~~~~~~~~
    at [redacted]/bug.ts:1:4

% deno
Deno 1.44.2
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> "".isWellFormed()
true
> 
```

Here is a workaround:

```
interface ExtraStringMethods {
  isWellFormed(): boolean;
}

function isWellFormed(str: string): boolean {
  return (str as unknown as ExtraStringMethods).isWellFormed();
}
```

### PR DESCRIPTION
**Pull request by dsherret**
Closes #24238
Closes https://github.com/denoland/deno/issues/23357

### CODE REVIEW

**Review by marvinhagemeister — APPROVED** [review]
LGTM

File: cli/tsc/dts/lib.dom.d.ts [hunk 1]
```diff
@@ -3151,8 +3152,6 @@ interface Blob {
     readonly type: string;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/arrayBuffer) */
     arrayBuffer(): Promise<ArrayBuffer>;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/bytes) */
-    bytes(): Promise<Uint8Array>;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/slice) */
```
Comment (dsherret) [inline cli/tsc/dts/lib.dom.d.ts:3155]:
```suggestion
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/bytes) */
    bytes(): Promise<Uint8Array>;
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/slice) */
```

File: cli/tsc/dts/lib.dom.d.ts [hunk 2]
```diff
@@ -3188,8 +3187,6 @@ interface Body {
     arrayBuffer(): Promise<ArrayBuffer>;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/blob) */
     blob(): Promise<Blob>;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/bytes) */
-    bytes(): Promise<Uint8Array>;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/formData) */
```
Comment (dsherret) [inline cli/tsc/dts/lib.dom.d.ts:3190]:
```suggestion
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/bytes) */
    bytes(): Promise<Uint8Array>;
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/formData) */
```

File: cli/tsc/dts/lib.webworker.d.ts [hunk 3]
```diff
@@ -5407,7 +5359,7 @@ declare var StylePropertyMapReadOnly: {
 };
 
 /**
- * This Web Crypto API interface provides a number of low-level cryptographic functions. It is accessed via the Crypto.subtle properties available in a window context (via globalThis.crypto).
```
Comment (bartlomieju) [inline cli/tsc/dts/lib.webworker.d.ts:5410]: I think this should not be changed?

File: cli/tsc/dts/lib.webworker.d.ts [hunk 4]
```diff
@@ -1025,8 +1020,6 @@ interface Blob {
     readonly type: string;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/arrayBuffer) */
     arrayBuffer(): Promise<ArrayBuffer>;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/bytes) */
-    bytes(): Promise<Uint8Array>;
```
Comment (bartlomieju) [inline cli/tsc/dts/lib.webworker.d.ts:1029]: This one should be kept

File: cli/tsc/dts/lib.webworker.d.ts [hunk 5]
```diff
@@ -1049,8 +1042,6 @@ interface Body {
     arrayBuffer(): Promise<ArrayBuffer>;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/blob) */
     blob(): Promise<Blob>;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/bytes) */
-    bytes(): Promise<Uint8Array>;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/formData) */
```
Comment (bartlomieju) [inline cli/tsc/dts/lib.webworker.d.ts:1045]: Ditto

**Review by bartlomieju — COMMENTED** [review]
Possibly fixes https://github.com/denoland/deno/issues/23357?

**Comment by imcotton** [comment]
Can you also include fixing for `Array.fromAsync` from microsoft/TypeScript#59837?

It only got merged last week thus not picked by 5.6, but should be fixed asap I think.

**Comment by petamoriken** [comment]
This PR fixes #24238

I am trying to migrate the `String#{is,to}WellFormed` type to ES2024 (https://github.com/microsoft/TypeScript/pull/58573). Anyway it has already been introduced as an ESNext type.

**Pull request merged**